From 6d0ec50cfbbceb4136cd38b19bb96c95a02507b8 Mon Sep 17 00:00:00 2001
From: Christopher Lam <calamity@chromium.org>
Date: Mon, 25 Feb 2019 06:52:25 +0000
Subject: [PATCH] [Bookmark Manager] Fix drag and drop onto input elements.

Bug: 931211
Change-Id: I29ad1489d1018c468df7d3178c0f9adae5f30aa8
Reviewed-on: https://chromium-review.googlesource.com/c/1484392
Reviewed-by: Alan Cutter <alancutter@chromium.org>
Commit-Queue: calamity <calamity@chromium.org>
Cr-Commit-Position: refs/heads/master@{#635036}
---
 chrome/browser/resources/bookmarks/dnd_manager.js | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/chrome/browser/resources/bookmarks/dnd_manager.js b/chrome/browser/resources/bookmarks/dnd_manager.js
index a859b99683a3b..3c1c4faf6ba2b 100644
--- a/chrome/browser/resources/bookmarks/dnd_manager.js
+++ b/chrome/browser/resources/bookmarks/dnd_manager.js
@@ -422,6 +422,11 @@ cr.define('bookmarks', function() {
      * @param {!Event} e
      */
     onDrop_(e) {
+      // Allow normal DND on text inputs.
+      if (e.path[0].tagName == 'INPUT') {
+        return;
+      }
+
       e.preventDefault();
 
       if (this.dropDestination_) {
@@ -454,10 +459,6 @@ cr.define('bookmarks', function() {
      * @param {Event} e
      */
     onDragOver_(e) {
-      // The default operation is to allow dropping links etc to do
-      // navigation. We never want to do that for the bookmark manager.
-      e.preventDefault();
-
       this.dropDestination_ = null;
 
       // Allow normal DND on text inputs.
@@ -465,6 +466,10 @@ cr.define('bookmarks', function() {
         return;
       }
 
+      // The default operation is to allow dropping links etc to do
+      // navigation. We never want to do that for the bookmark manager.
+      e.preventDefault();
+
       if (!this.dragInfo_.isDragValid()) {
         return;
       }
